File: doc/codes/python/examples/api/command/phase-correction.py
from autd3.commands import SetPhaseCorrection
from autd3.geometry import Autd3, Geometry
from autd3.value import Phase

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

phases = [[Phase.ZERO] * geometry.device(i).num_transducers() for i in range(geometry.num_devices())]

# ANCHOR: api
SetPhaseCorrection(phases)
# ANCHOR_END: api

File: doc/codes/rust/examples/api/command/phase-correction.rs
use autd3_rs::commands::SetPhaseCorrection;
use autd3_rs::geometry::{Autd3, Geometry};
use autd3_rs::value::Phase;

fn main() {
    let geometry = Geometry::new(vec![Autd3::default()]);

    let phases: Vec<Vec<Phase>> = geometry
        .iter()
        .map(|dev| vec![Phase::ZERO; dev.num_transducers()])
        .collect();

    // ANCHOR: api
    SetPhaseCorrection { phases: &phases };
    // ANCHOR_END: api
}
